src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -768,13 +770,17 @@ protected void populateEntity(Consumer entity, ConsumerDTO dto) {
             entity.setReleaseVer(new Release(dto.getReleaseVer().getReleaseVer()));
         }
 
-        if (dto.getEnvironment() != null) {
-            Environment env = environmentCurator.get(dto.getEnvironment().getId());
-            if (env == null) {
-                throw new NotFoundException(i18n.tr("Environment \"{0}\" could not be found.",
-                    dto.getEnvironment().getId()));
+        if (dto.getEnvironments() != null && dto.getEnvironments().size() > 0) {
```

Comment:
`.isEmpty` is more concise than `size() > 0`

Reply:
Done.